This short example explains how to read a .mat file generated by Octave or MATLAB®.<br/>
A test file containing coordinates from a drifting buoy is available in the data directory.

In [24]:
matfile = "../data/136014.mat"

The [File IO](http://docs.scipy.org/doc/scipy-0.16.1/reference/io.html) module has functions to read, write and inspect this type of file.

In [25]:
import scipy.io as sio

## Inspection

We use the function [whosmat](http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.io.whosmat.html#scipy.io.whosmat) to get information on the content:

In [26]:
sio.whosmat(matfile)

[('Lat', (847, 1), 'double'),
 ('Lon', (847, 1), 'double'),
 ('Lat_e', (847, 1), 'double'),
 ('Lon_e', (847, 1), 'double'),
 ('Time', (847, 1), 'double'),
 ('Drogue', (847, 1), 'double'),
 ('u', (847, 1), 'double'),
 ('v', (847, 1), 'double'),
 ('u_e', (847, 1), 'double'),
 ('v_e', (847, 1), 'double'),
 ('time_inizio', (1, 1), 'double'),
 ('time_fine', (1, 1), 'double')]

The file contains 12 variables (double), 10 of them are vectors of 847 elements.

## Reading

We use the function [loadmat](http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.io.loadmat.html):

In [27]:
data = sio.loadmat(matfile)
print type(data)

<type 'dict'>


The returned variable **data** is a dictionary, from which we can get the list of keys:

In [28]:
data.viewkeys()

dict_keys(['u_e', 'Lat_e', 'v', 'Lon', '__header__', '__globals__', 'v_e', 'Drogue', 'time_inizio', 'Lon_e', 'Time', 'Lat', 'u', '__version__', 'time_fine'])

and also access the different variables (first five elements here, to keep it short).

In [29]:
data['Time'][:5]

array([[ 735744.79166667],
       [ 735744.91666667],
       [ 735745.04166667],
       [ 735745.16666667],
       [ 735745.29166667]])

In [30]:
data['Lon'][:5]

array([[-0.8903565 ],
       [-0.85592446],
       [-0.8068571 ],
       [-0.7504404 ],
       [-0.70534913]])

## Other formats

Note that the IO module allows you to read other formats, such as IDL® files or Fortran files.